source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -262,10 +426,48 @@ void InstanceProfileCredentialsProvider::extractCredentials(
             secret_access_key.empty() ? "" : "*****", AWS_SESSION_TOKEN,
             session_token.empty() ? "" : "*****");
 
-  cached_credentials_ = Credentials(access_key_id, secret_access_key, session_token);
   last_updated_ = api_.timeSource().systemTime();
+  if (!Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.use_libcurl_to_fetch_aws_credentials") &&
+      context_) {
```

Comment:
Would be nice to make a helper function `usingAsyncHttp()` for this check since it's repeated a lot and not very readable with the negation.

Reply:
good idea, I will try to get these addressed asap before Matt gets a change to look.